### ISSUE TITLE
[assemble] JPackage does not register custom launcher

### ISSUE THREAD

**Issue by aalmiray** [jreleaser/jreleaser#840]
The `jpackage` assembler has an option to configure custom launchers which is not used at all. The log shows that `-add-launcher` is never passed as an argument as it should be.

### Environment Information

- **Operating System**: N/A
- **JReleaser Version:** 1.0.0
- **JDK Version:** N/A

**Comment by Shawoll** [comment]
Hi @aalmiray, how can we test any part of the `core/jreleaser-assemblers`? I didn't seen any `test` folder in there (I would like to create one, if that's a good idea). Maybe newbie question, but I didn't seen a lot of tests in the project at all; is it intended?

**Comment by aalmiray** [comment]
Hi @Shawoll, that is right. The project is light on unit tests at the moment. Performing tests on assemblers, tool processors, announcers, etc takes a lot of effort and coordination. For the time being these kind of tests are performed manually (not ideal in the long run). There are also some smoke tests (review files inside .github/workflows) that can tell immediately if something is horribly broken on the surface as there are no assertions at the moment. We've been able to catch plenty of errors in this way but of course more automation would be preferred. 

For the use case in point you'd have to ensure that the jpackage configuration created the extra launcher scripts. For this you need a Java project, configure jlink & jpackage, run the assembly, check the file system for a given file then checks its contents. That, or mock a lot of stuff which I think would be the wrong way to go.

**Comment by Shawoll** [comment]
Thanks. Of course, I've started working on this issue (as much as my family time allowed me :) ) and I like working on it. Yeah, the steps to reproduce this were little challenging to me, since I needed to configure JLink and other stuff, but I am making a nice progress and soon enough, I should have the pr ready for the review. I get it now, good starting issue: requires a knowledge of the system and synthetizes a lot of interesting features. I'll stay in touch. Thanks for the previous reply.
